hawc/apps/lit/managers.py

```diff
@@ -295,6 +295,41 @@ def with_tag(self, tag, descendants: bool = False):
             tag_ids.extend(list(tag.get_descendants().values_list("pk", flat=True)))
         return self.filter(tags__in=tag_ids).distinct("pk")
 
+    def require_tags(self, tags):
+        """
+        Filter references by any of the given tags.
+
+        Mirrors the filtering done in tagtree visuals.
+
+        Args:
+            tags: Required tags.
+        """
+        query = Q()
+        for tag in tags:
+            query |= Q(tags__in=[tag, *tag.get_descendants()])
+        return self.filter(query).distinct("pk")
+
+    def prune_tags(self, root_tag, pruned_tags):
```

Comment:
I think this is the reason we are getting different results.

The visual tagtree doesn't show the excluded nodes from the tree visual, but I don't think it filters references that have any of the tags in them. The help text currently says:

> Select tags which should be hidden from the tagtree. If a parent-tag is selected, all child-tags will also be hidden.

I think maybe we dont need this method here, and we should update the help text to state that it doesn't filter references which may include these tags?